### PR TITLE
Jjp241 oled block

### PR DESCRIPTION
**Pull request by jjp241**
Please merge it xd

### CODE REVIEW

File: code/oled_block/oled.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        if len(chunk) == line_width and chunk[-1] != ' ':
+            chunk = chunk + '-' # adds some swaggy hyphens when we split a word between two lines
+        draw.text((0,y_pos), chunk, font=font, fill = 255)
+        y_pos += 10
+
+    # Display image.
+    disp.image(image)
+    disp.display()
+
+while(True):
```
Comment (evemorgen) [inline code/oled_block/oled.py]: Python is not C, it does not require to use braces in this context :P

File: code/oled_block/oled.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import Adafruit_GPIO.SPI as SPI
+import Adafruit_SSD1306
+import time
+
+from PIL import Image
+from PIL import ImageDraw
+from PIL import ImageFont
+
+oled_buffer_path = 'oled_buffer.txt' # Path to oled buffer
+line_width = 20 # Number of characters in one line
```
Comment (evemorgen) [inline code/oled_block/oled.py]: I believe these couple variables are actually constants, in Python we use `CAPITAL_LETTERS` for constants

File: code/oled_block/oled.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import Adafruit_GPIO.SPI as SPI
```
Comment (evemorgen) [inline code/oled_block/oled.py]: why not `from Adafruit_GPIO import SPI`?

File: code/oled_block/README.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# OLED_BLOCK
+
+### Get this bad boy working:  
+Just follow [this](https://www.raspberrypi-spy.co.uk/2018/04/i2c-oled-display-module-with-raspberry-pi/) tutorial.
+
+### How it works?
+Program should have access to oled_buffer.txt file, you can specify the path in the code.
```
Comment (evemorgen) [inline code/oled_block/README.md]: if it's configurable, changes in code shouldn't be the default way of interaction with an user, please use [argparse](https://docs.python.org/3/library/argparse.html) or something similar

File: code/oled_block/README.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# OLED_BLOCK
```
Comment (evemorgen) [inline code/oled_block/README.md]: +1 for readme :)

File: code/oled_block/README.md (hunk 4 above)
Comment (evemorgen) [reply code/oled_block/README.md]: default values for these arguments are also required

File: code/oled_block/oled.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+def write_message(message):
+    disp.clear() #clear last message
+    disp.display()
+    y_pos = 0 #variable to store y axis position of line
+    image = Image.new('1', (width, height)) #our graphic with text
+
+    draw = ImageDraw.Draw(image) # Get drawing object to draw xd
+
+    split_string = [message[i:i+line_width] for i in xrange(0, len(message), line_width)]
+    for chunk in split_string[0:]:
```
Comment (evemorgen) [inline code/oled_block/oled.py]: what is the purpose of this `split_string[0:]`, I don't see the point of that O.o

File: code/oled_block/oled.py (hunk 1 above)
Comment (jjp241) [reply code/oled_block/oled.py]: K m8

File: code/oled_block/oled.py (hunk 2 above)
Comment (jjp241) [reply code/oled_block/oled.py]: Oke bruh

File: code/oled_block/oled.py (hunk 3 above)
Comment (jjp241) [reply code/oled_block/oled.py]: Now i  think that we don't need this library, we are not using spi to communicate with OLED

File: code/oled_block/oled.py (hunk 6 above)
Comment (jjp241) [reply code/oled_block/oled.py]: Like why those '[]'? Don't know for now, copied that, gonna check later if we can get rid of them

File: code/oled_block/README.md (hunk 4 above)
Comment (jjp241) [reply code/oled_block/README.md]: added config file :smile_cat:

File: code/oled_block/oled.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+def write_message(message):
+    disp.clear() #clear last message
+    disp.display()
+    y_pos = 0 #variable to store y axis position of line
+    image = Image.new('1', (width, height)) #our graphic with text
+
+    draw = ImageDraw.Draw(image) # Get drawing object to draw xd
+
+    split_string = [message[i:i+line_width] for i in xrange(0, len(message), line_width)]
```
Comment (evemorgen) [inline code/oled_block/oled.py]: I believe this suppose to be Python3 script but yet here we have `xrange`, which is Python2 specific

File: code/oled_block/oled.py (hunk 6 above)
Comment (evemorgen) [reply code/oled_block/oled.py]: Consider the following example:
we've got a list with elements: `list = [0, 1, 2, 3, 4, 5]`, square bracket operator with colon inside slices the list (gets specified part of the list), in this case `list[1:3]` will return `[1, 2]`. So when slicing from element `0` (like in the like 41) we'll get same thing `list[0:] == [0, 1, 2, 3, 4, 5]`. That's why I don't see the point of that

Comment (jjp241) [reply code/oled_block/oled.py]: You Wright

Comment (jjp241) [reply code/oled_block/oled.py]: Resolved in commit: 855a8c5

File: code/oled_block/oled.py (hunk 7 above)
Comment (jjp241) [reply code/oled_block/oled.py]: Resolved in commit: 855a8c5

**Pull request merged**